ldk/javascript/src/webpack/generate-banner.test.ts

```diff
@@ -0,0 +1,57 @@
+import { generateBanner } from "./generate-banner";
+import { LdkSettings, Ldk } from "./ldk-settings";
+
+function getLoopMetadataContent(encodedData: string): string {
+    const metadataContents = 
+        encodedData.replace('---BEGIN-LOOP-JSON-BASE64---', '')
+            .replace('---END-LOOP-JSON-BASE64---', '')
+            .replace('/*', '')
+            .replace('*/', '')
+            .trim();
+    return Buffer.from(metadataContents, 'base64').toString('utf-8');
+}
+
+describe('Generate Banner', () => {
+    it('generates banner given valid LdkSettings', () => {
+        const ldkSettings: LdkSettings = {
+            ldk: {
+                permissions: {
+                    clipboard: {},
+                    cursor: {},
+                    filesystem: { pathGlobs: [ { value: "/my/path" } ] },
+                    keyboard: {},
+                    network: { urlDomains: [ { value: "*.google.com" } ] },
+                    process: {},
+                    ui: {},
+                    vault: {},
+                    whisper: {},
+                    window: {},
+                }
+            }
+        }
+        const result = getLoopMetadataContent(generateBanner(ldkSettings));
+
+        //* aptitudes *//
+        expect(result).toContain("\"clipboard\":{}");
```

Comment:
Wondering whether it makes sense to test for string presence as opposed to deserialize JSON and testing deep object equality. This test structure would pass this data structure:
```yaml
ldk:
  permissions:
    unexpectedProperty: 
      clipboard: {}
```

Reply:
I like this idea. Quick enough to change it.